lib/kernel-manager.coffee

```diff
@@ -4,15 +4,26 @@ child_process = require 'child_process'
 Kernel = require './kernel'
 
 module.exports = KernelManager =
+    availableKernels:  []
     runningKernels: {}
-
-    getAvailableKernels: _.memoize ->
-        try
-            out = child_process.spawnSync('jupyter',['kernelspec','list', '--json']).stdout.toString()
-        catch
-            out = child_process.spawnSync('ipython',['kernelspec','list', '--json']).stdout.toString()
-        _.pluck(JSON.parse(out).kernelspecs, 'spec')
-
+    
+    getAvailableKernels: ->
+        @availableKernels = @getConfigJson 'kernelspecs', @availableKernels
+        @updateKernels() unless @availableKernels.length
+        @availableKernels
+               
+    updateKernels: ->
+        save = (out) =>
+            @availableKernels = _.pluck JSON.parse(out).kernelspecs, 'spec'
```

Comment:
Yeah we need tooling for upgrading from old specs to new specs, though I think it can be a follow on PR after.